### ISSUE TITLE
Shadcn support

### ISSUE THREAD

**Issue by JElgar** [EvHaus/react-pdf-charts#531]
### Describe the bug

I'm trying to use this library with a shadcn chart. Im getting `Your chart rendered a node type (style) which isn't supported by <ReactPDFChart /> yet`. Not sure if this is a shadcn addition or something in recharts (or just me doing crazy things) but would be awesome if this library supported shadcn charts!

### Reproduction

Any of the shadcn example seem to have the same issue https://ui.shadcn.com/charts. E.g.

```tsx
import React from 'react';
import { Page, Text, Document, PDFViewer } from '@react-pdf/renderer';
import ReactPDFChart from 'react-pdf-charts';
import { FundWithProvider } from '@/lib/providers/types';
import { Bar, BarChart, CartesianGrid, XAxis } from 'recharts';

import {
	ChartConfig,
	ChartContainer,
} from '@/components/ui/chart';

const chartData = [
	{ month: 'January', desktop: 186 },
	{ month: 'February', desktop: 305 },
	{ month: 'March', desktop: 237 },
	{ month: 'April', desktop: 73 },
	{ month: 'May', desktop: 209 },
	{ month: 'June', desktop: 214 },
];

const chartConfig = {
	desktop: {
		label: 'Desktop',
		color: 'hsl(var(--chart-1))',
	},
} satisfies ChartConfig;

export function MyDocument() {
	return <PDFViewer className="h-full">
		<Document>
			<Page size="A4">
				<ReactPDFChart>
					<ChartContainer config={chartConfig}>
						<BarChart accessibilityLayer data={chartData}>
							<CartesianGrid vertical={false} />
							<XAxis
								dataKey="month"
								tickLine={false}
								tickMargin={10}
								axisLine={false}
								tickFormatter={(value) => value.slice(0, 3)}
							/>
							<Bar dataKey="desktop" fill="var(--color-desktop)" radius={8} />
						</BarChart>
					</ChartContainer>
				</ReactPDFChart>
			</Page>
		</Document>
	</PDFViewer>;
}
```

### Validations

- [X] Check that there isn't [already an issue](https://github.com/EvHaus/react-pdf-charts/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that you're confident that this is a legitimate bug. To ask a question open a [GitHub Discussion](https://github.com/EvHaus/react-pdf-charts/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by PieterT2000** [comment]
As far as I understand, you can't wrap the Recharts component in a Shadcn ChartContainer since the library expects a fixed width and height for your Barchart component. The ChartContainer wraps the bar chart inside a ResponsiveContainer, which doesn't work. Other reasons are related to Providers, context, and how this library renders React components. Shadcn's ChartContainer wraps the Rechart component in a provider, but since this library renders everything statically, it will break if it encounters a `useSomeContext.` hook call.

**Comment by EvHaus** [comment]
The error means your code is generating `<style>` tag somewhere inside the `<ReactPDFChart>` component. Pieter might be correct about it being inside the `<ChartContainer>` component.

If you could create a minimal reproduction by cloning [this StackBlitz](https://stackblitz.com/edit/react-pdf-charts-starter?file=src%2FApp.tsx) I can try to help you troubleshoot it further from there.

**Comment by PieterT2000** [comment]
Indeed, a `<ChartStyle />` component (which renders a style tag) gets rendered inside the `<ChartContainer>` component.
Checkout the [source code here](https://github.com/shadcn-ui/ui/blob/1081536246b44b6664f4c99bc3f1b3614e632841/apps/www/registry/default/ui/chart.tsx#L60).

**Comment by JElgar** [comment]
> I am trying to put simple Barchart in PDF but bars not getting printed for me, is this a bug
You need to disable animation for the Bar components (`isAnimationActive={false}`) https://github.com/EvHaus/react-pdf-charts?tab=readme-ov-file#known-issues

**Comment by JElgar** [comment]
Hmm ok so it sounds like my question is just a bad one. Most of what shadcn adds on top of recharts can't really be applied to pdfs (web styling e.g. for tooltips). I.e. if you want to use this library with shadcn charts, just don't, use rechart chart directly. Is that the right understanding?
